custom_model_runner/datarobot_drum/drum/model_adapter.py

```diff
@@ -320,12 +334,18 @@ def predict(self, input_filename, model=None, **kwargs):
         if self._custom_hooks.get(CustomHooks.TRANSFORM):
             try:
                 # noinspection PyCallingNonCallable
+                if self._target_type == TargetType.TRANSFORM:
+                    nrow_original = data.shape[0]
                 data = self._custom_hooks[CustomHooks.TRANSFORM](data, model)
+
             except Exception as exc:
                 raise type(exc)(
                     "Model transform hook failed to transform dataset: {}".format(exc)
                 ).with_traceback(sys.exc_info()[2]) from None
             self._validate_data(data, CustomHooks.TRANSFORM)
+            if self._target_type == TargetType.TRANSFORM:
+                self._validate_transform_rows(nrow_original, data)
+                return data
```

Comment:
I don't like this early return here

Reply:
I'm open to that- just bypass predict entirely if it's a transform target type...let me take a look